src/sema/expression/arithmetics.rs

```diff
@@ -0,0 +1,864 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::sema::ast::{Expression, Namespace, RetrieveType, StringLocation, Type};
```

Comment:
To me arithmetics sounds strange, I don't really know of this being used in the plural. `arithmetic.rs` might be better.

There is a pedantic point about bitwise operations not being arithmetic, but I don't think that matters.

Reply:
You're right. The word has no `s` in the end. I can call this file `mathematics.rs`. Does this improve anything?

Reply:
To be pedantic, it's not math, it's simple arithmetic.

Reply:
> To be pedantic, it's not math, it's simple arithmetic.

Arithmetic is also math.